test/unit/BPool/BPool_SwapExactAmountIn.tree

```diff
@@ -0,0 +1,26 @@
+BPool::SwapExactAmountIn
+├── when reentrancy lock is set
+│   └── it should revert
+├── when pool is not finalized
+│   └── it should revert
+├── when token in is not bound
+│   └── it should revert
+├── when token out is not bound
+│   └── it should revert
+├── when token in exceeds max allowed ratio
+│   └── it should revert
+├── when spot price before swap exceeds maxPrice
+│   └── it should revert
+├── when spot price after swap exceeds maxPrice
+│   └── it should revert
+├── when returned tokenOut is less than minAmountOut
+│   └── it should revert
+└── when preconditions are met
+    ├── it emits a LOG_CALL event
+    ├── it sets the reentrancy lock
+    ├── it emits a LOG_SWAP event
+    ├── it calls _pullUnderlying for tokenIn
```

Comment:
the user not having enough `tokenIn` would revert on `_pullUnderlying`, perhaps it's better test modularization to not test that flow here, but with `_pullUnderlying`? tree has less branches this way, but it's less descriptive of domain cases and isn't consistent with what I just did on `exitPool`